### PR TITLE
Comments Endpoint

### LINKED ISSUE #21: Author 008-b: Create Basic Stream Post
User should be able to create a basic post
title
source
origin
description
content type
content
author
categories
published
id
visibility
visibleTo: [] <- not supported yet
unlisted

Refer to API guide on github for details of attributes.

### LINKED ISSUE #135: backend comments

### PR DESCRIPTION
**Pull request by htruong4**
resolves #135  #153

### CODE REVIEW

**Comment by shchau** [comment]
Worked for me 👍

**Review by hiufungk — CHANGES_REQUESTED** [review]
waiting to have a common way of getting post id with other groups

File: backend/api/view/PostCommentsView.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        #     return False
+        return True
+
+    def insert_local_comment(self, request):
+        post_data = request.data["post"]
+        post_data_split = post_data.split("/")
+        # from server
+        post_short_id = post_data_split[-1]
+        local_post_filter = Post.objects.filter(id=str(post_short_id))
+        if(local_post_filter.exists()):
```
Comment (forgeno) [inline backend/api/view/PostCommentsView.py:35]: Can you please add error checking

File: backend/api/view/PostCommentsView.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            author_post = local_post_filter[0]
+
+            author_data = request.data["comment"]["author"]["id"].split("/")
+
+            if(can_read(str(author_data[-1]), PostSerializer(author_post).data)):
+                Comment.objects.create(
+                            author=request.data["comment"]["author"]["id"],
```
Comment (forgeno) [inline backend/api/view/PostCommentsView.py]: can you modulize `request.data["comment"]["author"]` and others

File: backend/api/view/PostCommentsView.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        if(not self.valid_payload(request.data)):
+            return Response("Error: Payload does not match", status.HTTP_400_BAD_REQUEST)
+        else:
+            author_profile_filter = AuthorProfile.objects.filter(user=request.user)
+            server_user_exists = ServerUser.objects.filter(user=request.user).exists()
+
+            if(author_profile_filter.exists()):
+                # request is from front end
+                author_profile = author_profile_filter[0]
+                payload_author_id = request.data["comment"]["author"]["id"]
```
Comment (forgeno) [inline backend/api/view/PostCommentsView.py:85]: modulize this as well

**Review by forgeno — CHANGES_REQUESTED** [review]

**Review by hiufungk — APPROVED** [review]

**Review by forgeno — APPROVED** [review]

File: backend/api/view/StreamPostsView.py [hunk 4]
```diff
@@ -52,9 +52,70 @@ def get(self, request):
...
                 sorted_comments= sorted(post["comments"], key=lambda k: k['published'], reverse=True)
-                post["comments"] = sorted_comments
-                if (can_read(user_id, post)):
-                    stream.append(post)
+                comments = []
+                for comment in sorted_comments:
+                    parsed_post_url = urlparse(comment["author"])
+                    commenter_host = '{}://{}/'.format(parsed_post_url.scheme, parsed_post_url.netloc)
+                    #check if the user is a local author
+                    if (can_read(user_id, post)):
```
Comment (hiufungk) [inline backend/api/view/StreamPostsView.py]: check can read before preparing comments

**Review by hiufungk — CHANGES_REQUESTED** [review]

**Review by hiufungk — APPROVED** [review]

**Pull request merged**